### ISSUE TITLE
scipy.optimize.minimize ignoring the optional argument 'maxiter'

### ISSUE THREAD

**Issue by tianBreznik** [scipy/scipy#9141]
<<Please describe the issue in detail here, and for bug reports fill in the fields below.>>
The function call optimize.minimize(costFunction, x0 = NN.initial_nn_params, options = {'disp':True,'maxiter':500}, method="L-BFGS-B", jac=True) appears to be ignoring the maxiter constraint and iterates a significantly less number of times. I have reproduced the same neural network in Matlab and set its number of iterations to 500 and it takes significantly longer than python which leads me to believe that minimize is not iterating 500 times. Additionally I have set maxiter to a ridiculous number: 1000000 and the execution time was the same as when I set it to 10. I have also tried to run it with maxfun and the same issue appears.

I would appreciate any advice on this issue.

Thank you!

### Reproducing code example:
```
optimize.minimize(costFunction, x0 = NN.initial_nn_params, options = {'disp':True,'maxfun':maxiter}, method="L-BFGS-B", jac=True)
```
### Error message:
```
<<Full error message, if any (starting from line Traceback: ...)>>
```

### Scipy/Numpy/Python version information:
```
('1.1.0', '1.14.5', sys.version_info(major=2, minor=7, micro=10, releaselevel='final', serial=0))

```

**Comment by pv** [comment]
maxiter is the maximum allowed iteration number, end reaching it results.
The algorithm terminates earlier if the termination tolerance is satisfied.
To control termination, the termination tolerance should be adjusted (is not necessarily the same as on matlab).

**Comment by tianBreznik** [comment]
But in Matlab, using the same function and exactly the same structure of the data, I get a success rate of 96% from my neural network predictions if I set the iterations to 500. But if I set the number of iterations to 1, I get a success rate of roughly 75% which is almost exactly the same I get with scipy.optimize.minimize after the termination tolerance is satisfied, as you explained. So is there a way for me to change this tolerance so that it will not terminate with the iterations as quickly?

Thank you, I appreciate your help!
